go-controller/pkg/ovn/controller/apbroute/external_controller_policy.go

```diff
@@ -207,13 +207,13 @@ func (m *externalPolicyManager) updateRoutePolicy(existingPolicy *routePolicySta
 				existingTargetPodConfig, found := existingNs[updatedPodNamespacedName]
 				if !found {
 					existingTargetPodConfig = newPodInfo()
-					existingNs[updatedPodNamespacedName] = existingTargetPodConfig
 				}
 				// applyPodConfig will apply changes and update existingTargetPodConfig with the status
 				err := m.applyPodConfig(updatedPod, existingTargetPodConfig, updatedPolicy)
 				if err != nil {
 					return err
 				}
+				existingNs[updatedPodNamespacedName] = existingTargetPodConfig
```

Comment:
because it otherwise adds the pod information before it is processed by the network client. If there is an error while processing, as it is in the case of recreating the same pod  in the existing code, the cached information from the NS shows as the pod information has been updated, when in fact it has not because of the error.

Reply:
yes, and it will have `failed` flag set to true, and we need that information to clean up later in case a part of the config was applied (e.g. static gateways were applied, but dynamic failed). Did it cause some problems during testing?

Reply:
Yeah, it failed when I was running the new unit test to reproduce the pod re-create issue: the test always succeeded, the comparison between structures did not fail.

Reply:
that is worth investigating

Reply:
it looks like when updating/adding it does not flag as `failed`: there is no logic to handle the error as it is for the delete use case:
https://github.com/ovn-org/ovn-kubernetes/pull/3841/files#diff-5bcf37e71dfd7a40592da7f85186b7ed97c49b962d4787753c56b54ad3e942f5R172-R177